packages/lisk-transaction-pool/src/transaction_pool.ts

```diff
@@ -13,89 +13,159 @@
 }
 
-interface ProcessTransactionsResponse {
-	readonly errors: ReadonlyArray<Error>;
-	readonly invalidTransactions: ReadonlyArray<Transaction>;
-	readonly validTransactions: ReadonlyArray<Transaction>;
-}
-
-type processTransactions = (
-	transactions: ReadonlyArray<Transaction>,
-) => ProcessTransactionsResponse;
-
 export class TransactionPool {
-	private applyTransactions: processTransactions;
-	private queues: Queues;
-	private validateTransactions: processTransactions;
-	private verifyTransactions: processTransactions;
-
-	public constructor(
-		validateTransactions: processTransactions,
-		verifyTransactions: processTransactions,
-		applyTransactions: processTransactions,
-	) {
-		this.queues = {
+	// tslint:disable-next-line variable-name
+	private readonly _queues: Queues;
+
+	public constructor() {
+		this._queues = {
 			received: new Queue(),
 			validated: new Queue(),
 			verified: new Queue(),
 			pending: new Queue(),
 			ready: new Queue(),
 		};
-		this.validateTransactions = validateTransactions;
-		this.verifyTransactions = verifyTransactions;
-		this.applyTransactions = applyTransactions;
 	}
 
-	public addTransactions(transactions: ReadonlyArray<Transaction>): void {}
+	public addTransactions(transactions: ReadonlyArray<Transaction>): void {
+		transactions.forEach((transaction: Transaction) => {
+			if (this.existsInTransactionPool(transaction)) {
+				this._queues.received.enqueueOne(transaction);
+			}
+		});
+	}
 
-	public getProcessableTransactions(
-		limit: number,
-	): ReadonlyArray<Transaction> {}
+	public addVerifiedRemovedTransactions(transactions: ReadonlyArray<Transaction>): void {
+		const { received, validated, ...otherQueues } = this._queues;
 
-	public onDeleteBlock(block: Block): void {}
+		// Move transactions from the verified, pending and ready queues to the validated queue where account was a receipient in the verified removed transactions
+		const removedTransactionsByRecipientId = this.removeTransactionsFromQueues(
+			Object.keys(otherQueues),
+			queueCheckers.checkTransactionForRecipientId(transactions),
+		);
 
-	public onNewBlock(block: Block): void {}
+		this._queues.validated.enqueueMany(removedTransactionsByRecipientId);
+		// Add transactions to the verfied queue which were included in the verified removed transactions
+		this._queues.verified.enqueueMany(transactions);
+	}
 
-	public onRoundRollback(delegates: ReadonlyArray<string>): void {}
+	public existsInTransactionPool(transaction: Transaction): boolean {
+		return Object.keys(this._queues).reduce(
+			(previousValue, currentValue) =>
+				previousValue || this._queues[currentValue].exists(transaction),
+			false,
+		);
+	}
 
-	public verifyTransaction(): void {}
+	public get queues(): Queues {
+		return this._queues;
+	}
 
-	private existsInTransactionPool(transaction: Transaction): boolean {}
+	public getProcessableTransactions(limit: number): ReadonlyArray<Transaction> {
+		return this._queues.ready.dequeueUntil(
```

Comment:
Wasnt the idea to keep it in the queue until we `removeConfirmedTransaction`?

Reply:
This function is not just the basic implementation, I'll update this function I implement this feature.